diff --git a/node_modules/@mattermost/react-native-emm/ios/ScreenCaptureManager.swift b/node_modules/@mattermost/react-native-emm/ios/ScreenCaptureManager.swift
index 46ec43c..9227372 100644
--- a/node_modules/@mattermost/react-native-emm/ios/ScreenCaptureManager.swift
+++ b/node_modules/@mattermost/react-native-emm/ios/ScreenCaptureManager.swift
@@ -163,11 +163,11 @@
     }
     
     @objc func applyBlurEffect(notification: Notification) {
-        self.applyBlurEffect(radius: 10.0)
+        self.applyBlurEffect(radius: 20.0)
     }
 
     
-    @objc public func applyBlurEffect(radius: Double = 10.0) {
+    @objc public func applyBlurEffect(radius: Double = 20.0) {
         if self.blurView == nil && (
                 (self.preventScreenCapture && !self.isAuthenticating) ||
                 (self.isAuthenticating && self.blurOnAuthenticate)
