diff --git a/node_modules/@mattermost/react-native-network-client/android/build.gradle b/node_modules/@mattermost/react-native-network-client/android/build.gradle
index 5541e06..ad7391d 100644
--- a/node_modules/@mattermost/react-native-network-client/android/build.gradle
+++ b/node_modules/@mattermost/react-native-network-client/android/build.gradle
@@ -131,9 +131,4 @@ dependencies {
   testImplementation "junit:junit:4.12"
   testImplementation "org.mockito:mockito-core:3.+"
   testImplementation "com.squareup.okhttp3:mockwebserver:4.12.0"
-
-  debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}")
-  debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
-    exclude group:'com.squareup.okhttp3', module:'okhttp'
-  }
 }
diff --git a/node_modules/@mattermost/react-native-network-client/android/src/debug/kotlin/com/mattermost/networkclient/NetworkClient.kt b/node_modules/@mattermost/react-native-network-client/android/src/debug/kotlin/com/mattermost/networkclient/NetworkClient.kt
index d4a6559..76c62a0 100644
--- a/node_modules/@mattermost/react-native-network-client/android/src/debug/kotlin/com/mattermost/networkclient/NetworkClient.kt
+++ b/node_modules/@mattermost/react-native-network-client/android/src/debug/kotlin/com/mattermost/networkclient/NetworkClient.kt
@@ -1,9 +1,6 @@
 package com.mattermost.networkclient
 
-import com.facebook.flipper.plugins.network.FlipperOkhttpInterceptor
-import com.facebook.flipper.plugins.network.NetworkFlipperPlugin
 import com.facebook.react.bridge.*
-import com.mattermost.networkclient.interceptors.*
 import okhttp3.*
 import java.net.URI
 
@@ -21,20 +18,4 @@ internal class NetworkClient(baseUrl: HttpUrl? = null, options: ReadableMap? = n
 
         okHttpClient = builder.build()
     }
-
-    override fun applyGenericClientBuilderConfiguration() {
-        super.applyGenericClientBuilderConfiguration()
-        applyFlipperInterceptor()
-    }
-
-    override fun applyClientBuilderConfiguration(options: ReadableMap?, cookieJar: CookieJar?) {
-        super.applyClientBuilderConfiguration(options, cookieJar)
-        applyFlipperInterceptor()
-    }
-
-    private fun applyFlipperInterceptor() {
-        if (RCTOkHttpClientFactory.flipperPlugin != null) {
-            builder.addNetworkInterceptor(FlipperOkhttpInterceptor(RCTOkHttpClientFactory.flipperPlugin as NetworkFlipperPlugin))
-        }
-    }
 }
