diff --git a/node_modules/@react-native-community/netinfo/android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java b/node_modules/@react-native-community/netinfo/android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java
index 252d1e7..09328c5 100644
--- a/node_modules/@react-native-community/netinfo/android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java
+++ b/node_modules/@react-native-community/netinfo/android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java
@@ -91,14 +91,14 @@ public class NetworkCallbackConnectivityReceiver extends ConnectivityReceiver {
             // Get the connection type
             if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_BLUETOOTH)) {
                 connectionType = ConnectionType.BLUETOOTH;
+            } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_VPN)) {
+                connectionType = ConnectionType.VPN;
             } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
                 connectionType = ConnectionType.CELLULAR;
             } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
                 connectionType = ConnectionType.ETHERNET;
             } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
                 connectionType = ConnectionType.WIFI;
-            } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_VPN)) {
-                connectionType = ConnectionType.VPN;
             }
 
             if (network != null) {
diff --git a/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts b/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts
index 757d784..663abda 100644
--- a/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts
+++ b/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts
@@ -109,10 +109,10 @@ export default class InternetReachability {
           const nextTimeoutInterval = this._isInternetReachable
             ? this._configuration.reachabilityLongTimeout
             : this._configuration.reachabilityShortTimeout;
-          this._currentTimeoutHandle = setTimeout(
-            this._checkInternetReachability,
-            nextTimeoutInterval,
-          );
+          // this._currentTimeoutHandle = setTimeout(
+          //   this._checkInternetReachability,
+          //   nextTimeoutInterval,
+          // );
         },
       )
       .catch(
@@ -125,10 +125,10 @@ export default class InternetReachability {
             }
             
             this._setIsInternetReachable(false);
-            this._currentTimeoutHandle = setTimeout(
-              this._checkInternetReachability,
-              this._configuration.reachabilityShortTimeout,
-            );
+            // this._currentTimeoutHandle = setTimeout(
+            //   this._checkInternetReachability,
+            //   this._configuration.reachabilityShortTimeout,
+            // );
           }
         },
       )
