diff --git a/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts b/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts
index ed49fb7..23dc6ae 100644
--- a/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts
+++ b/node_modules/@react-native-community/netinfo/src/internal/internetReachability.ts
@@ -109,20 +109,20 @@ export default class InternetReachability {
           const nextTimeoutInterval = this._isInternetReachable
             ? this._configuration.reachabilityLongTimeout
             : this._configuration.reachabilityShortTimeout;
-          this._currentTimeoutHandle = setTimeout(
-            this._checkInternetReachability,
-            nextTimeoutInterval,
-          );
+          // this._currentTimeoutHandle = setTimeout(
+          //   this._checkInternetReachability,
+          //   nextTimeoutInterval,
+          // );
         },
       )
       .catch(
         (error: Error | 'timedout' | 'canceled'): void => {
           if (error !== 'canceled') {
             this._setIsInternetReachable(false);
-            this._currentTimeoutHandle = setTimeout(
-              this._checkInternetReachability,
-              this._configuration.reachabilityShortTimeout,
-            );
+            // this._currentTimeoutHandle = setTimeout(
+            //   this._checkInternetReachability,
+            //   this._configuration.reachabilityShortTimeout,
+            // );
           }
         },
       )
