diff --git a/node_modules/@react-native-community/art/android/build.gradle b/node_modules/@react-native-community/art/android/build.gradle
index 949bc00..ffeb4ff 100644
--- a/node_modules/@react-native-community/art/android/build.gradle
+++ b/node_modules/@react-native-community/art/android/build.gradle
@@ -3,10 +3,6 @@ buildscript {
     google()
     jcenter()
   }
-
-  dependencies {
-    classpath 'com.android.tools.build:gradle:3.2.1'
-  }
 }
 
 def getExtOrDefault(name) {
@@ -31,15 +27,15 @@ android {
 
 repositories {
   google()
-  jcenter()
   mavenCentral()
+  mavenLocal()
 }
 
 allprojects {
     repositories {
+        mavenCentral()
         mavenLocal()
         google()
-        jcenter()
         maven {
             url "$rootDir/../node_modules/react-native/android"
         }
