diff --git a/node_modules/@react-native-community/cli-platform-apple/build/commands/runCommand/getBuildSettings.js b/node_modules/@react-native-community/cli-platform-apple/build/commands/runCommand/getBuildSettings.js
index df09405..2c81e5d 100644
--- a/node_modules/@react-native-community/cli-platform-apple/build/commands/runCommand/getBuildSettings.js
+++ b/node_modules/@react-native-community/cli-platform-apple/build/commands/runCommand/getBuildSettings.js
@@ -48,7 +48,7 @@ async function getBuildSettings(xcodeProject, mode, buildOutput, scheme, target)
     }
   }
   const targetIndex = applicationTargets.indexOf(selectedTarget);
-  return settings[targetIndex].buildSettings;
+  return settings.find(s => s.target === selectedTarget).buildSettings;
 }
 function getPlatformName(buildOutput) {
   // Xcode can sometimes escape `=` with a backslash or put the value in quotes
