diff --git a/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h b/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h
index f9020d1..7962e7b 100644
--- a/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h
+++ b/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h
@@ -1,9 +1,5 @@
 
-#if __has_include("RCTBridgeModule.h")
-#import "RCTBridgeModule.h"
-#else
 #import <React/RCTBridgeModule.h>
-#endif
 
 @interface RNRudderSdkModule : NSObject <RCTBridgeModule>
 
