diff --git a/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle b/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle
index 1536175..e4dd282 100644
--- a/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle
+++ b/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle
@@ -15,7 +15,7 @@ android {
     compileSdkVersion 29
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 21
         targetSdkVersion 29
         versionCode 1
         versionName "1.0"
diff --git a/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h b/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h
index f9020d1..d2cf7c9 100644
--- a/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h
+++ b/node_modules/@rudderstack/rudder-sdk-react-native/ios/RNRudderSdkModule.h
@@ -1,6 +1,6 @@
 
-#if __has_include("RCTBridgeModule.h")
-#import "RCTBridgeModule.h"
+#if __has_include(<React/RCTBridgeModule.h>)
+#import <React/RCTBridgeModule.h>
 #else
 #import <React/RCTBridgeModule.h>
 #endif
