diff --git a/node_modules/@sentry/react-native/sentry.gradle b/node_modules/@sentry/react-native/sentry.gradle
index c54bf8c..33364e9 100644
--- a/node_modules/@sentry/react-native/sentry.gradle
+++ b/node_modules/@sentry/react-native/sentry.gradle
@@ -134,7 +134,7 @@ gradle.projectsEvaluated {
                       "--url", sentryProps.get("defaults.url"),
                       "--auth-token", sentryProps.get("auth.token")
               ])
-              args.addAll(["react-native", "gradle",
+              args.addAll(["react-native", "gradle","--debug",
                            "--bundle", bundleOutput,           // The path to a bundle that should be uploaded.
                            "--sourcemap", sourcemapOutput,     // The path to a sourcemap that should be uploaded.
                            "--release", releaseName,            // The name of the release to publish.
