diff --git a/node_modules/@sentry/react-native/sentry.gradle b/node_modules/@sentry/react-native/sentry.gradle
index c54bf8c..e76a028 100644
--- a/node_modules/@sentry/react-native/sentry.gradle
+++ b/node_modules/@sentry/react-native/sentry.gradle
@@ -127,8 +127,8 @@ gradle.projectsEvaluated {
               //
               def args = [cliExecutable]
 
-              args.addAll(!config.logLevel ? [] : [
-                      "--log-level", config.logLevel      // control verbosity of the output
+              args.addAll([
+ "--log-level", "debug"      // control verbosity of the output
               ])
               args.addAll(!config.flavorAware ? [] : [
                       "--url", sentryProps.get("defaults.url"),
