diff --git a/node_modules/@stream-io/flat-list-mvcp/android/build.gradle b/node_modules/@stream-io/flat-list-mvcp/android/build.gradle
index 1cad8fd..b0b564f 100644
--- a/node_modules/@stream-io/flat-list-mvcp/android/build.gradle
+++ b/node_modules/@stream-io/flat-list-mvcp/android/build.gradle
@@ -29,7 +29,7 @@ android {
   compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
   buildToolsVersion getExtOrDefault('buildToolsVersion')
   defaultConfig {
-    minSdkVersion 16
+    minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 21
     targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
     versionCode 1
     versionName "1.0"
