diff --git a/node_modules/@types/commonmark/index.d.ts b/node_modules/@types/commonmark/index.d.ts
index 35e9ed6..382cf98 100755
--- a/node_modules/@types/commonmark/index.d.ts
+++ b/node_modules/@types/commonmark/index.d.ts
@@ -5,8 +5,8 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 export type NodeType =
-    'text' |'softbreak' | 'linebreak' | 'emph' | 'strong' | 'html_inline' | 'link' | 'image' | 'code' | 'document' | 'paragraph' |
-    'block_quote' | 'item' | 'list' | 'heading' | 'code_block' | 'html_block' | 'thematic_break' | 'custom_inline' | 'custom_block';
+    'text' |'softbreak' | 'linebreak' | 'emph' | 'strong' | 'html_inline' | 'link' | 'image' | 'code' | 'document' | 'paragraph' | 'mention_highlight' | 'at_mention' |
+    'block_quote' | 'item' | 'list' | 'heading' | 'code_block' | 'html_block' | 'thematic_break' | 'custom_inline' | 'custom_block' | 'table' | 'edited_indicator';
 
 export class Node {
     constructor(nodeType: NodeType, sourcepos?: Position);
@@ -125,6 +125,8 @@ export class Node {
      * https://github.com/jgm/commonmark.js/issues/74
      */
     _listData: ListData;
+
+    mentionName?: string;
 }
 
 /**
@@ -200,6 +202,8 @@ export interface ParserOptions {
      */
     smart?: boolean | undefined;
     time?: boolean | undefined;
+    urlFilter?: (url: string) => boolean;
+    minimumHashtagLength?: number;
 }
 
 export interface HtmlRenderingOptions extends XmlRenderingOptions {
