diff --git a/node_modules/@types/react-syntax-highlighter/index.d.ts b/node_modules/@types/react-syntax-highlighter/index.d.ts
index 6ee9af3..cf1516e 100755
--- a/node_modules/@types/react-syntax-highlighter/index.d.ts
+++ b/node_modules/@types/react-syntax-highlighter/index.d.ts
@@ -33,6 +33,10 @@ declare module 'react-syntax-highlighter' {
     export { default as Prism } from 'react-syntax-highlighter/dist/esm/prism';
 }
 
+declare module 'react-syntax-highlighter/create-element' {
+    export {createStyleObject} from 'react-syntax-highlighter/create-element';
+}
+
 // esm start
 declare module 'react-syntax-highlighter/dist/esm/default-highlight' {
     import * as React from 'react';
