diff --git a/node_modules/expo-modules-core/build/web/CoreModule.js b/node_modules/expo-modules-core/build/web/CoreModule.js
index 7a02851..b1b9561 100644
--- a/node_modules/expo-modules-core/build/web/CoreModule.js
+++ b/node_modules/expo-modules-core/build/web/CoreModule.js
@@ -1,4 +1,4 @@
-import uuid from '../uuid';
+import * as uuid from 'uuid';
 class EventEmitter {
     listeners;
     addListener(eventName, listener) {
