diff --git a/node_modules/expo-video-thumbnails/ios/VideoThumbnailsModule.swift b/node_modules/expo-video-thumbnails/ios/VideoThumbnailsModule.swift
index 2b01a1d..78c7ffa 100644
--- a/node_modules/expo-video-thumbnails/ios/VideoThumbnailsModule.swift
+++ b/node_modules/expo-video-thumbnails/ios/VideoThumbnailsModule.swift
@@ -7,7 +7,7 @@ public class VideoThumbnailsModule: Module {
   public func definition() -> ModuleDefinition {
     Name("ExpoVideoThumbnails")
 
-    AsyncFunction("getThumbnail", getVideoThumbnail).runOnQueue(.main)
+      AsyncFunction("getThumbnail", getVideoThumbnail).runOnQueue(.global())
   }
 
   internal func getVideoThumbnail(sourceFilename: URL, options: VideoThumbnailsOptions) throws -> [String: Any] {
