diff --git a/node_modules/react-intl/src/components/createIntl.js b/node_modules/react-intl/src/components/createIntl.js
index 6887909..d1819bc 100644
--- a/node_modules/react-intl/src/components/createIntl.js
+++ b/node_modules/react-intl/src/components/createIntl.js
@@ -32,7 +32,7 @@ var formatMessage = function (config, formatters, descriptor, rawValues) {
         formatters,
         descriptor,
         values], rest, false));
-    return (0, utils_1.toKeyedReactNodeArray)(chunks);
+    return typeof chunks === 'string' ? chunks : (0, utils_1.toKeyedReactNodeArray)(chunks);
 };
 /**
  * Create intl object
