diff --git a/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js b/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js
index 4ab9d7e..67b2060 100644
--- a/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js
+++ b/node_modules/react-native-reanimated/lib/reanimated2/js-reanimated/index.web.js
@@ -11,7 +11,7 @@ export const _updatePropsJS = (_viewTag, _viewName, updates, viewRef) => {
             acc[index][key] = value;
             return acc;
         }, [{}, {}]);
-        setNativeProps(viewRef._component, rawStyles);
+        setNativeProps(viewRef.current._component, rawStyles);
     }
 };
 const setNativeProps = (component, style) => {
diff --git a/node_modules/react-native-reanimated/src/reanimated2/js-reanimated/index.web.ts b/node_modules/react-native-reanimated/src/reanimated2/js-reanimated/index.web.ts
index 46de206..58126f2 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/js-reanimated/index.web.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/js-reanimated/index.web.ts
@@ -18,7 +18,7 @@ export const _updatePropsJS = (_viewTag, _viewName, updates, viewRef) => {
       [{}, {}]
     );
 
-    setNativeProps(viewRef._component, rawStyles);
+    setNativeProps(viewRef.current._component, rawStyles);
   }
 };
 
