diff --git a/node_modules/react-native-fast-image/dist/index.d.ts b/node_modules/react-native-fast-image/dist/index.d.ts
index 0b1afd5..2226715 100644
--- a/node_modules/react-native-fast-image/dist/index.d.ts
+++ b/node_modules/react-native-fast-image/dist/index.d.ts
@@ -89,6 +89,10 @@ export interface FastImageProps extends AccessibilityProps {
      * Render children within the image.
      */
     children?: React.ReactNode;
+    /**
+     * Native ID to used with shared elements transition
+     */
+    nativeID?: string;
 }
 interface FastImageStaticProperties {
     resizeMode: typeof resizeMode;
