diff --git a/node_modules/react-native-slider/lib/Slider.js b/node_modules/react-native-slider/lib/Slider.js
index c640410..dc5c971 100644
--- a/node_modules/react-native-slider/lib/Slider.js
+++ b/node_modules/react-native-slider/lib/Slider.js
@@ -5,6 +5,7 @@ var _react=require("react");var _react2=_interopRequireDefault(_react);
 
 
 var _reactNative=require("react-native");
+var _styles=require("deprecated-react-native-prop-types");
 
 
 
@@ -549,13 +550,13 @@ return false;}}]);return Slider;}(_react.PureComponent);Slider.propTypes={ /**
      * the slider is released).
      */onSlidingComplete:_propTypes2.default.func, /**
      * The style applied to the slider container.
-     */style:_reactNative.ViewPropTypes.style, /**
+     */style:_styles.ViewPropTypes.style, /**
      * The style applied to the track.
-     */trackStyle:_reactNative.ViewPropTypes.style, /**
+     */trackStyle:_styles.ViewPropTypes.style, /**
      * The style applied to the thumb.
-     */thumbStyle:_reactNative.ViewPropTypes.style, /**
+     */thumbStyle:_styles.ViewPropTypes.style, /**
      * Sets an image for the thumb.
-     */thumbImage:_reactNative.Image.propTypes.source, /**
+     */thumbImage:_styles.ImagePropTypes.source, /**
      * Set this to true to visually see the thumb touch rect in green.
      */debugTouchArea:_propTypes2.default.bool, /**
      * Set to true to animate values with default 'timing' animation type
diff --git a/node_modules/react-native-slider/src/Slider.js b/node_modules/react-native-slider/src/Slider.js
index 37deee5..514a057 100644
--- a/node_modules/react-native-slider/src/Slider.js
+++ b/node_modules/react-native-slider/src/Slider.js
@@ -10,9 +10,9 @@ import {
   StyleSheet,
   PanResponder,
   View,
-  Easing,
-  ViewPropTypes
+  Easing
 } from "react-native";
+import {ViewPropTypes} from 'deprecated-react-native-prop-types';
 
 import PropTypes from 'prop-types';
 
