diff --git a/node_modules/react-native-hw-keyboard-event/android/build.gradle b/node_modules/react-native-hw-keyboard-event/android/build.gradle
index a70aace..d2acc75 100644
--- a/node_modules/react-native-hw-keyboard-event/android/build.gradle
+++ b/node_modules/react-native-hw-keyboard-event/android/build.gradle
@@ -4,7 +4,7 @@ android {
     compileSdkVersion 28
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 16
         targetSdkVersion 28
         versionCode 1
         versionName "1.0"
