diff --git a/node_modules/react-native-haptic-feedback/android/build.gradle b/node_modules/react-native-haptic-feedback/android/build.gradle
index c326a26..87f4814 100644
--- a/node_modules/react-native-haptic-feedback/android/build.gradle
+++ b/node_modules/react-native-haptic-feedback/android/build.gradle
@@ -20,7 +20,7 @@ android {
     buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 16
         targetSdkVersion rootProject.hasProperty('targetSdkVersion') ? rootProject.targetSdkVersion : DEFAULT_TARGET_SDK_VERSION
         versionCode 1
         versionName "1.0"
diff --git a/node_modules/react-native-haptic-feedback/android/src/main/java/com/mkuczera/RNReactNativeHapticFeedbackModule.java b/node_modules/react-native-haptic-feedback/android/src/main/java/com/mkuczera/RNReactNativeHapticFeedbackModule.java
index cc597e0..db3c3f9 100644
--- a/node_modules/react-native-haptic-feedback/android/src/main/java/com/mkuczera/RNReactNativeHapticFeedbackModule.java
+++ b/node_modules/react-native-haptic-feedback/android/src/main/java/com/mkuczera/RNReactNativeHapticFeedbackModule.java
@@ -39,7 +39,7 @@ public class RNReactNativeHapticFeedbackModule extends ReactContextBaseJavaModul

     switch (type) {
       case "impactLight":
-        durations = new long[]{0, 20};
+        durations = new long[]{0, 5};
         break;
       case "impactMedium":
         durations = new long[]{0, 40};
