--- a/node_modules/react-native-keyboard-controller/src/animated.tsx
+++ b/node_modules/react-native-keyboard-controller/src/animated.tsx
@@ -260,9 +260,9 @@ export const KeyboardProvider = (props: KeyboardProviderProps) => {
       <KeyboardControllerViewAnimated
         ref={viewRef}
         enabled={enabled}
-        navigationBarTranslucent={IS_EDGE_TO_EDGE || navigationBarTranslucent}
-        statusBarTranslucent={IS_EDGE_TO_EDGE || statusBarTranslucent}
-        preserveEdgeToEdge={IS_EDGE_TO_EDGE || preserveEdgeToEdge}
+        navigationBarTranslucent={true}
+        statusBarTranslucent={true}
+        preserveEdgeToEdge={true}
         style={styles.container}
         // on*Reanimated prop must precede animated handlers to work correctly
         onKeyboardMoveReanimated={keyboardHandler}
