diff --git a/node_modules/react-native-localize/android/src/main/java/com/zoontek/rnlocalize/RNLocalizeModule.java b/node_modules/react-native-localize/android/src/main/java/com/zoontek/rnlocalize/RNLocalizeModule.java
index 73fcd14..8c9e447 100644
--- a/node_modules/react-native-localize/android/src/main/java/com/zoontek/rnlocalize/RNLocalizeModule.java
+++ b/node_modules/react-native-localize/android/src/main/java/com/zoontek/rnlocalize/RNLocalizeModule.java
@@ -48,7 +48,6 @@ public class RNLocalizeModule extends ReactContextBaseJavaModule implements Life
   private final List<String> USES_RTL_LAYOUT =
     Arrays.asList("ar", "ckb", "fa", "he", "ks", "lrc", "mzn", "ps", "ug", "ur", "yi");
 
-  private @Nullable final BroadcastReceiver mBroadcastReceiver;
   private boolean mMainActivityVisible = true;
   private boolean mEmitChangeOnResume = false;
 
@@ -62,7 +61,7 @@ public class RNLocalizeModule extends ReactContextBaseJavaModule implements Life
     filter.addAction(Intent.ACTION_TIME_CHANGED);
     filter.addAction(Intent.ACTION_TIMEZONE_CHANGED);
 
-    mBroadcastReceiver = new BroadcastReceiver() {
+    BroadcastReceiver receiver = new BroadcastReceiver() {
       @Override
       public void onReceive(Context context, Intent intent) {
         if (intent.getAction() != null) {
@@ -72,7 +71,7 @@ public class RNLocalizeModule extends ReactContextBaseJavaModule implements Life
     };
 
     reactContext.addLifecycleEventListener(this);
-    reactContext.registerReceiver(mBroadcastReceiver, filter);
+    reactContext.registerReceiver(receiver, filter);
   }
 
   @Override
@@ -112,9 +111,7 @@ public class RNLocalizeModule extends ReactContextBaseJavaModule implements Life
   }
 
   @Override
-  public void onHostDestroy() {
-    getReactApplicationContext().unregisterReceiver(mBroadcastReceiver);
-  }
+  public void onHostDestroy() {}
 
   private void emitLocalizationDidChange() {
     if (getReactApplicationContext().hasActiveCatalystInstance()) {
