diff --git a/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceTypeResolver.java b/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceTypeResolver.java
index a8e9b0c..2fac079 100644
--- a/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceTypeResolver.java
+++ b/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceTypeResolver.java
@@ -80,10 +80,10 @@ public class DeviceTypeResolver {
     double heightInches = metrics.heightPixels / (double) metrics.ydpi;
     double diagonalSizeInches = Math.sqrt(Math.pow(widthInches, 2) + Math.pow(heightInches, 2));
 
-    if (diagonalSizeInches >= 3.0 && diagonalSizeInches <= 6.9) {
+    if (diagonalSizeInches >= 3.0 && diagonalSizeInches <= 7.9) {
       // Devices in a sane range for phones are considered to be Handsets.
       return DeviceType.HANDSET;
-    } else if (diagonalSizeInches > 6.9 && diagonalSizeInches <= 18.0) {
+    } else if (diagonalSizeInches > 7.9 && diagonalSizeInches <= 18.0) {
       // Devices larger than handset and in a sane range for tablets are tablets.
       return DeviceType.TABLET;
     } else {
