diff --git a/node_modules/react-native-reanimated/android/expo/linking.gradle b/node_modules/react-native-reanimated/android/expo/linking.gradle
index 3a583a6..81ae08b 100644
--- a/node_modules/react-native-reanimated/android/expo/linking.gradle
+++ b/node_modules/react-native-reanimated/android/expo/linking.gradle
@@ -15,12 +15,24 @@ if (isExpoLinked) {
     apply plugin: 'com.android.library'
 
     android {
+        // Disable aar bundling to fix the error when running `gradle assembleRelease` over `gradle :app:assembleRelease`
+        // which will throw `Direct local .aar file dependencies are not supported when building an AAR.`
+        tasks.whenTaskAdded { task ->
+            if (['bundleDebugAar', 'bundleReleaseAar'].contains(task.name)) {
+                task.enabled = false
+            }
+        }
+
         compileSdkVersion safeExtGet('compileSdkVersion', 30)
 
         sourceSets {
             main.manifest.srcFile 'src/AndroidManifest.xml'
             main.java.srcDirs = [ 'expo/src/main/java' ]
         }
+
+        libraryVariants.all {
+            generateBuildConfigProvider?.get()?.enabled = false
+        }
     }
 
     dependencies {
